compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -162,6 +169,22 @@ class FirExpressionsResolveTransformer(transformer: FirBodyResolveTransformer) :
         return completeInference.compose()
     }
 
+    // Use the generic invoke method in Function classes, to match bridges generated by the backend.
+    private fun checkIfInvoke(functionCall: FirFunctionCall) : FirFunctionCall {
```

Comment:
I'd think frontend shouldn't rewrite declarations in sake of knowledge of backend behavior, so this logic should rather be in fir2ir or even in backend itself

Reply:
moved to fir2ir